src/parser/ts/index.ts

```diff
@@ -15,9 +21,19 @@ const compiledGrammar = Grammar.fromCompiled(grammar);
 export const formQueriesForExpression = async (
   expression: string,
   target: QueryTarget,
-  connector: Connector
+  connector: Connector,
+  defaultInterval: string
 ): Promise<DataQueryRequestItem[]> => {
-  const parsed = parse(expression);
+  const rawParsed = parse(expression);
+  const defaultGranularity = target.granularity || defaultInterval;
+  const defaultAggregation = target.aggregation;
```

Comment:
maybe we just pass all this stuff (`target. granularity, granularity.aggregatetion, interval`) to the `injectAggregatesToParsed ` and hide logic there?
same with `defaultAggregation === 'none'`

Reply:
end call it `enrichWithDefaultAggregates` or something

Reply:
so we don't bloat the main function

Reply:
hmm, it makes sense, I'll redo this as well